### DISCUSSION TITLE
PowerShell environment variables and filenames in Actions

### DISCUSSION THREAD

**Discussion by ndarilek** [community/community#25923] [Actions]
<p>I have an action that needs to generate a UWP package. Things worked fine when my commands were plain and didn’t use any environment variables. I.e.:</p>
<pre><code class="lang-auto">      - run: |
          ...
          makeappx pack /d myproject /p myproject.appx
</code></pre>
<p>Now, in an effort to make this action more generic, I’ve done:</p>
<pre><code class="lang-auto">      - run: echo ::set-env name=REPOSITORY_NAME::$(echo "$GITHUB_REPOSITORY" | awk -F / '{print $2}' | sed -e "s/:refs//")
        shell: bash
</code></pre>
<p>Then:</p>
<pre><code class="lang-auto">makeappx pack /d ${env:REPOSITORY_NAME} /p ${env:REPOSITORY_NAME}.appx
</code></pre>
<p>When doing that, I get an error about the “/p” option not being recognized. I suspect a command line parsing bug where perhaps a missing parameter reports as an unknown option, so I do:</p>
<pre><code class="lang-auto">move package.appx ${env:REPOSITORY_NAME}.appx
</code></pre>
<p>after generating package.appx. As suspected, <code>move</code> complains that the destination is null.</p>
<p>Interestingly, I have:</p>
<pre><code class="lang-auto">          makeappx unpack /p uwp\${env:REPOSITORY_NAME}.appx /d ${env:REPOSITORY_NAME}
</code></pre>
<p>earlier in my action, and it works just fine. So there’s something with my use of <code>${env:REPOSITORY_NAME}.appx</code> later that doesn’t work. Also, <code>$REPOSITORY_NAME</code> works fine on my Linux jobs, so I know that it is getting set correctly.</p>
<p>What am I doing wrong here? This is probably more of a generic PowerShell issue, but it seems like I’m using environment variables correctly, and I can’t find the right Google keywords to fix this.</p>
<p>Thanks.</p>

**Comment by dogboydog** [comment, marked as answer]
<p>Hi,</p>
<p>I’m not sure what exactly is causing your issue, but can you try this syntax instead and see if it works:</p>
<pre><code class="lang-auto">move package.appx ${{ env.REPOSITORY_NAME  }}.appx
</code></pre>
<p>This lets GitHub interpolate the env for you</p>

**Comment by ndarilek** [comment]
<p>Doh, didn’t even think about that. Yeah, that seems to have fixed it.<br>
Thanks!</p>
